docs/custom_promise_types/groups.py

```diff
@@ -0,0 +1,345 @@
+import re
+import json
+from subprocess import Popen, PIPE
+from cfengine import PromiseModule, ValidationError, Result
+
+
+class GroupsPromiseTypeModule(PromiseModule):
+    def __init__(self):
+        super().__init__("git_promise_module", "0.0.2")
+        self._name_regex = re.compile(r"^[a-z_][a-z0-9_-]*[$]?$")
+        self._name_maxlen = 32
+
+
+    def validate_promise(self, promiser, attributes):
+        # check promiser type
+        if type(promiser) is not str:
+            raise ValidationError("invalid type for promiser: expected string")
```

Comment:
I think all your error messages and log messages should be changed to have Stor forbokstav ™

Reply:
Stor forbokstav ™ it is